avocado/utils/archive.py

```diff
@@ -176,15 +177,73 @@ class ArchiveFile:
 
     # extension info: is_zip, is_tar, zipfile|tarfile, +mode
     _extension_table = {
+        # ZIP archives
         ".zip": (True, False, zipfile.ZipFile, ""),
+        # TAR archives (uncompressed)
         ".tar": (False, True, tarfile.open, ""),
+        # TAR archives with gzip compression
         ".tar.gz": (False, True, tarfile.open, ":gz"),
         ".tgz": (False, True, tarfile.open, ":gz"),
+        # TAR archives with bzip2 compression
         ".tar.bz2": (False, True, tarfile.open, ":bz2"),
         ".tbz2": (False, True, tarfile.open, ":bz2"),
+        # TAR archives with xz compression
+        ".tar.xz": (False, True, tarfile.open, ":xz"),
+        ".txz": (False, True, tarfile.open, ":xz"),
         ".xz": (False, True, tarfile.open, ":xz"),
+        # TAR archives with zstd compression
+        ".tar.zst": (False, True, tarfile.open, ":zstd"),
+        ".tzst": (False, True, tarfile.open, ":zstd"),
+        # Standalone compressed files (not tar archives)
+        ".gz": (False, False, None, "gz"),
+        ".bz2": (False, False, None, "bz2"),
+        ".zst": (False, False, None, "zst"),
     }
 
+    @staticmethod
+    def _detect_archive_type(filename):
+        """
+        Detect archive type based on file content.
+
+        :param filename: the archive file name.
+        :return: tuple of (is_zip, is_tar, engine, extra_mode) or None if not an archive.
+        """
+        result = None
+
+        # Check for ZIP file
+        if zipfile.is_zipfile(filename):
+            result = (True, False, zipfile.ZipFile, "")
+        # Check for TAR file
+        elif tarfile.is_tarfile(filename):
+            # Detect compression method for tar files
+            with open(filename, "rb") as f:
```

Comment:
Do you think the logic here could be transformed into a data structure or even added to the existing (or a modified version) of the `_extension_table` dict?